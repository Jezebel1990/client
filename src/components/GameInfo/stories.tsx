import { Meta, StoryObj } from '@storybook/react'
import GameInfo, { GameInfoProps} from '.' 
import mockGame from './mock'

export default { 
    title: 'Game/GameInfo', 
    component: GameInfo,
    parameters: {
        backgrounds: {
          default: 'won-dark'
        }
      },
      args: mockGame
} as Meta 

export const Default: StoryObj<GameInfoProps> = (args: any) => (
    <div style={{ maxWidth: '144rem', margin: 'auto', padding: '1.5rem' }}>
      <GameInfo {...args} />
    </div>
  ) 

  Default.args = {}