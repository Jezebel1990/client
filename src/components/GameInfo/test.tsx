import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import GameInfo from '.'

const props = {
  title: 'My Game Title',
  description: 'Game Description',
  price: '210,00',
}

describe('<GameInfo />', () => {
  it('should render game informations', () => {
    const { container } = renderWithTheme(<GameInfo {...props} />)

    screen.debug() 

    expect(
      screen.getByRole('heading', { name: /my game title/i })
    ).toBeInTheDocument()
    
    expect(screen.getByText(/210,00/)).toBeInTheDocument()
    expect(screen.getByText(/game description/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
