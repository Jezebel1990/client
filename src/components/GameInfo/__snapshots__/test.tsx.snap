// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfo /> should render game informations 1`] = `
.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff77b0 0%,#a618df 50%);
  color: #FAFAFA;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c6:hover {
  background: linear-gradient(180deg, #ec146e 0%, #5c057e 50% );
}

.c6 svg {
  width: 1.5rem;
}

.c6 svg+span {
  margin-left: 0.8rem;
}

.c8 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff77b0 0%,#a618df 50%);
  color: #FAFAFA;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  background: none;
  color: #6d47cc;
}

.c8:hover {
  background: none;
}

.c8 svg {
  width: 1.5rem;
}

.c8 svg+span {
  margin-left: 0.8rem;
}

.c8:hover {
  color: #5531af;
}

.c1 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c1::after {
  position: absolute;
  left: 0;
  bottom: -0.5rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #13D7E0;
}

.c3 {
  position: absolute;
  top: 1.6rem;
  display: flex;
  align-items: center;
  font-weight: 600;
  color: #FAFAFA;
  background-color: #13D7E0;
  font-size: 1.4rem;
  padding: 0 2.4rem;
  height: 3.6rem;
  right: -2rem;
}

.c3::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c3::before {
  border-left-color: #0b7d82;
  border-top-color: #0b7d82;
}

.c3::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

.c0 {
  position: relative;
  background: #FAFAFA;
  padding: 2.4rem;
}

.c0 .c2 {
  right: -1rem;
}

.c0 .c2:before {
  border-right-width: 1rem;
}

.c4 {
  font-size: 1.4rem;
  color: #18181b;
  margin-bottom: 2.4rem;
}

.c5 {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.c5 >button {
  width: 100%;
  margin-bottom: 0.8rem;
}

@media (min-width: 768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media (min-width: 768px) {
  .c0 .c2 {
    right: 2.4rem;
    top: 2.4rem;
    font-size: 1.8rem;
  }

  .c0 .c2:before {
    border: none;
  }
}

@media (min-width: 768px) {
  .c4 {
    max-width: 77rem;
  }
}

@media (min-width: 768px) {
  .c5 {
    flex-direction: row-reverse;
  }

  .c5 >button {
    width: initial;
    margin-bottom: 0;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="black"
    linecolor="primary"
  >
    My Game Title
  </h2>
  <div
    class="c2 c3"
    color="primary"
  >
    $210,00
  </div>
  <p
    class="c4"
  >
    Game Description
  </p>
  <div
    class="c5"
  >
    <button
      class="c6"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
        />
      </svg>
      <span>
        Add to cart
      </span>
    </button>
    <button
      class="c8"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m19.65 9.04-4.84-.42-1.89-4.45c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5 4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.73 3.67-3.18c.67-.58.32-1.68-.56-1.75zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
        />
      </svg>
      <span>
        Wishlist
      </span>
    </button>
  </div>
</div>
`;
